xmtp_mls/src/groups/mod.rs

```diff
@@ -522,8 +541,44 @@ where
     // Update this installation's leaf key in the group by creating a key update commit
     pub async fn key_update(&self) -> Result<(), GroupError> {
         let conn = &mut self.client.store.conn()?;
+        
+        self.pre_intent_hook().await.unwrap();
+
+        let intent = NewGroupIntent::new(IntentKind::KeyUpdate, self.group_id.clone(), vec![]);
+        intent.store(conn)?;
+
+        self.sync_with_conn(conn).await
+    }
+
+    // Update the installation's leaf key in a synchronous way.
+    pub fn queue_key_update(&self) -> Result<(), GroupError> {
+        let conn = &mut self.client.store.conn()?;
+        
+        self.pre_intent_hook();
+
         let intent = NewGroupIntent::new(IntentKind::KeyUpdate, self.group_id.clone(), vec![]);
         intent.store(conn)?;
+        Ok(())
+    }
+
+    /// Checking the last key rotation time before rotating the key. 
+    pub async fn pre_intent_hook(&self) -> Result<(), GroupError> {
+        let conn = &self.client.store.conn()?;
+        let last_rotated_time = conn.get_rotated_time_checked(self.group_id.clone());
+        
+        match last_rotated_time {
+            Ok(rotated_time) => {
+                if rotated_time == 0 {
+                    let _ = self.queue_key_update();
+                    let update_interval = Some(5_000_000);
+                    self.maybe_update_installations(conn, update_interval).await?;
+                }
+            }
+            Err(err) => {
+                // Handle the error appropriately
+                log::error!("database error fetching rotated time {:?}", err);
```

Comment:
You can probably just bubble the error up instead of using a match statement and then logging the error.

let last_rotated_time = conn.get_rotated_time_checked(self.group_id.clone())**?**;